File: game-server/src/main/java/net/dodian/utilities/Range.java
package net.dodian.utilities;

/**
 * @author Dashboard
 */
public class Range {

    private final int floor;
    private final int ceiling;

    public Range(int floor, int ceiling) {
        this.floor = floor;
        this.ceiling = ceiling;
    }

    public int getFloor() {
        return this.floor;
    }

    public int getCeiling() {
        return this.ceiling;
    }

    public int getValue() {
        return floor + (int) (Math.random() * (ceiling - floor + 1));
    }

}

File: game-server/src/main/kotlin/net/dodian/utilities/Range.kt
package net.dodian.utilities

import kotlin.jvm.JvmName

class Range(
    private val floorRaw: Int,
    private val ceilingRaw: Int,
) {
    fun getFloor(): Int = floorRaw

    @get:JvmName("getFloorValue")
    val floor: Int
        get() = getFloor()

    fun getCeiling(): Int = ceilingRaw

    @get:JvmName("getCeilingValue")
    val ceiling: Int
        get() = getCeiling()

    @get:JvmName("getValueProperty")
    val value: Int
        get() = getValue()

    fun getValue(): Int = floorRaw + (Math.random() * (ceilingRaw - floorRaw + 1)).toInt()
}
